backend/app/modules/user/budgets/service/upcoming.py

```diff
@@ -0,0 +1,20 @@
+from app.context.manager import Context
+from app.pydantic import Model
+from app.modules.modules_factory.protocols import SimulatePendingSnapshots
```

Comment:
I prefer adding protocols which are used by a service in a service file, then re-using the ones we have in `protocols.py`. This is because it makes the services independent so we won't end up with circular deps or we won't have to implement functions that we don't need, because some protocol forces as to do so.

I know that it adds some additional boilerplate code, so we can discuss it if you reckon it's not worth it

Reply:
that's ok, i've changed it - it's better to be consist than think now about potential change that'll touch another places in the codebase; nonetheless, your way of redefining protocols is reasonable :v: